library/src/main/java/io/ona/kujaku/KujakuLibrary.java

```diff
@@ -0,0 +1,109 @@
+package io.ona.kujaku;
+
+import android.content.Intent;
+import android.os.AsyncTask;
+import android.util.Log;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.List;
+
+import io.ona.kujaku.activities.MapActivity;
+import io.ona.kujaku.callables.AsyncTaskCallable;
+import io.ona.kujaku.domain.Point;
+import io.ona.kujaku.listeners.OnFinishedListener;
+import io.ona.kujaku.tasks.GenericAsyncTask;
+import io.ona.kujaku.utils.Constants;
+
+import static io.ona.kujaku.utils.IOUtil.readInputStreamAsString;
+
+/**
+ * @author Vincent Karuri
+ */
+public class KujakuLibrary {
+
+    private BaseKujakuApplication hostApplication;
```

Comment:
I think there are two ways to do this. 

One is to provide callbacks library methods and allow those to handle data returned from the library to the host application.

The second would be to provide a reference to a host application (within the library) which implements a library application interface that the library can use to send data to.

Here, I used the second approach so the reference variable is necessary.

When you say set during instantiation, do you suggest setting the `hostApplication` variable in  the getInstance method of the library?

Reply:
> When you say set during instantiation, do you suggest setting the hostApplication variable in the getInstance method of the library?

Yea

Would a third one be to use the `startActivityForResult` and return the points added back to the calling activity so that it can process them as it wishes. The assumption being that this activity(`MapActivity`) will always be started by another activity so that the style and other configurations can be provided to it

Reply:
I think the host application must be passed in a constructor or via indirectly via an init method and throwing an error if the initialization is has not been done and getInstance method is invoked.

Reply:
I agree with this approach, similar to stock library I think?